pkg/operator/controller/ingress/deployment.go

```diff
@@ -325,7 +325,34 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, ingressController
 	secretName := controller.RouterEffectiveDefaultCertificateSecretName(ci, deployment.Namespace)
 	deployment.Spec.Template.Spec.Volumes[0].Secret.SecretName = secretName.Name
 
+	var (
+		useSyslogSidecar  bool
+		useSyslogEndpoint bool
```

Comment:
These bools are mutually exclusive and seem to be proxies for `Spec.Logging.Access.Destination.Type`... what if instead you declare up front a LoggingDestinationType pointer which defaults to `nil` and is set to either Container or Syslog depending on annotations or logging config, and then finally switch on the destination rather than the booleans?

Reply:
Latest push does something along those lines, using `AccessLogging` to store the type and parameters (ensuring that the specified type's parameters field is non-nil).